pkg/apis/management.cattle.io/v3/authn_types.go

```diff
@@ -92,6 +92,7 @@ type UserAttribute struct {
 	GroupPrincipals map[string]Principals // the value is a []Principal, but code generator cannot handle slice as a value
 	LastRefresh     string
 	NeedsRefresh    bool
+	Extra           map[string]map[string][]string
```

Comment:
Wondering if `Extra` name with long type like `map[string]map[string][]string` is too generic for something that looks to me like extra fields per provider? Just looking at field name and type it's not clear that the key of the map is the authn provider name.
It was much more clear when type was `map[string][]string` but addition of extra layer makes it more obscure I think.
Or are there limitations for adding more types here?

Reply:
@snasovich that map[string][]string is as defined by Kubernetes struct to carry userInfo https://pkg.go.dev/k8s.io/apiserver@v0.20.2/pkg/authentication/user#DefaultInfo

And we need to carry this per authProvider hence another map of provider -> defaultInfo so that we do not have to translate it from some custom type when sending it downstream

Reply:
@prachidamle , thank you for the explanation. `map[string][]string` makes sense since it's what referenced in k8s struct as you linked. I still think the name `Extra` doesn't make it obvious what is used as the key, so maybe we can rename this field to something like `ExtraByProvider` or `ProviderExtra` - provided we can map it to Extra of type `map[string][]string` that is actually logged as "extra" field on audit log entry (I'm not sure what type is actually used to produce audit log entries)?

Reply:
I agree with changing the name as the other fields called `Extra` are for an individual provider, `ExtraByProvider` sounds fine. Can a comment be added as well explaining the field? That type in general is intimidating so having an explanation and example would help.

Reply:
Done